src/Carbon.ts

```diff
@@ -85,6 +91,17 @@ export class Class extends AbstractContext.Class {
 		return this.getDocumentMetadata<System.InstanceMetadata.Class>( "system.instance.metadata" );
 	}
 
+	connectMessaging( options:Messaging.Options, onConnect:() => void, onError:( error:Error ) => void ):void;
+	connectMessaging( onConnect:() => void, onError:( error:Error ) => void ):void;
+	connectMessaging( optionsOrOnConnect:Messaging.Options | ( () => void ), onConnectOrOnError:( () => void ) | ( ( error:Error ) => void ), onError?:( error:Error ) => void ):void {
```

Comment:
`connectMessaging` feels weird... `connect` would actually be worse... I think we may need to expose the messaging service so it makes more sense to people: `carbon.messaging.connect()`

Reply:
I agree, exposing the messaging service would be better for the user.